storage_controller/src/service.rs

```diff
@@ -1231,6 +1231,13 @@ impl Service {
         &self,
         attach_req: AttachHookRequest,
     ) -> anyhow::Result<AttachHookResponse> {
+        let _tenant_lock = trace_exclusive_lock(
+            &self.tenant_op_locks,
+            attach_req.tenant_shard_id.tenant_id,
+            TenantOperations::ShardSplit,
+        )
+        .await;
```

Comment:
Not mentioned in the PR description & seems orthogonal.

Reply:
This was an opportunistic fix for a rare test failure, which should become more common  in https://github.com/neondatabase/neon/pull/8135 (but the fix is still correct before that PR)